detect_secrets/plugins/keyword.py

```diff
@@ -55,7 +55,7 @@ def analyze_string(self, string, line_num, filename):
         if WHITELIST_REGEX.search(string):
             return output
 
-        for identifier in self.secret_generator(string.lower()):
+        for identifier in self.secret_generator(string):
```

Comment:
It's probably more performant to make the string lowered once here, and passed into `secret_generator`, than to make it `.lower()` every time.